subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -500,16 +500,29 @@ u8_t ll_adv_enable(u8_t enable)
 
 			ull_filter_adv_pdu_update(adv, rl_idx, pdu_adv);
 			ull_filter_adv_pdu_update(adv, rl_idx, pdu_scan);
+
+			/* In case the local IRK was not set or no match was
+			 * found the fallback address was used instead, check
+			 * that a valid address has been set.
+			 */
+			if (!mem_nz(&pdu_adv->adv_ind.addr[0], BDADDR_SIZE)) {
```

Comment:
This can be shared for both the privacy and non-privacy cases, please move.

Reply:
Moved. Extended advertising is different, so it will have to have it's own check I think. (for example anonymous advertising).